testutils/clients/pulse_client.go

```diff
@@ -37,5 +46,35 @@ func GetTestPulseClient(pn uint32, err error) *TestPulseClient {
 	client.topSyncPulse = func(ctx context.Context, in *exporter.GetTopSyncPulse, opts ...grpc.CallOption) (response *exporter.TopSyncPulseResponse, e error) {
 		return getTestTopSyncPulseResponse(pn), err
 	}
+	client.NextFinalizedPulseFunc = func(ctx context.Context, in *exporter.GetNextFinalizedPulse, opts ...grpc.CallOption) (*exporter.FullPulse, error) {
+		return GetFullPulse(pn), nil
+	}
 	return client
 }
+
+func GetFullPulse(pn uint32) *exporter.FullPulse {
+	time, err := insolar.PulseNumber(pn).AsApproximateTime()
+	if err != nil {
+		panic(err)
```

Comment:
pulse_client is a testutil, it's not a good idea to panic.
Either return err and handle it with testing.T or add  testing.T as param to the func and fail with T

(test that will panic has no logs and may lead to stopping next tests execution)